angular/projects/admin-nrpti/src/app/records/annual-reports/annual-report-add-edit/annual-report-add-edit.component.ts

```diff
@@ -0,0 +1,354 @@
+        ),
+        paragraph: new FormControl(
+          (this.currentRecord && this.currentRecord.legislation && this.currentRecord.legislation.paragraph) || ''
+        )
+      }),
+      legislationDescription: new FormControl((this.currentRecord && this.currentRecord.legislationDescription) || ''),
+      issuedTo: new FormGroup({
+        type: new FormControl(
+          (this.currentRecord && this.currentRecord.issuedTo && this.currentRecord.issuedTo.type) || ''
+        ),
+        companyName: new FormControl(
+          (this.currentRecord && this.currentRecord.issuedTo && this.currentRecord.issuedTo.companyName) || ''
+        ),
+        firstName: new FormControl(
+          (this.currentRecord && this.currentRecord.issuedTo && this.currentRecord.issuedTo.firstName) || ''
+        ),
+        middleName: new FormControl(
+          (this.currentRecord && this.currentRecord.issuedTo && this.currentRecord.issuedTo.middleName) || ''
+        ),
+        lastName: new FormControl(
+          (this.currentRecord && this.currentRecord.issuedTo && this.currentRecord.issuedTo.lastName) || ''
+        ),
+        fullName: new FormControl(
+          (this.currentRecord && this.currentRecord.issuedTo && this.currentRecord.issuedTo.fullName) || ''
+        ),
+        dateOfBirth: new FormControl(
+          (this.currentRecord &&
+            this.currentRecord.issuedTo &&
+            this.currentRecord.issuedTo.dateOfBirth &&
+            this.utils.convertJSDateToNGBDate(new Date(this.currentRecord.issuedTo.dateOfBirth))) ||
+          ''
+        ),
+        anonymous: new FormControl(
+          (this.currentRecord && this.currentRecord.issuedTo && this.currentRecord.issuedTo.anonymous) || ''
+        )
+      }),
+      projectName: new FormControl((this.currentRecord && this.currentRecord.projectName) || ''),
+      location: new FormControl((this.currentRecord && this.currentRecord.location) || ''),
+      latitude: new FormControl(
+        (this.currentRecord && this.currentRecord.centroid && this.currentRecord.centroid[1]) || ''
+      ),
+      longitude: new FormControl(
+        (this.currentRecord && this.currentRecord.centroid && this.currentRecord.centroid[0]) || ''
+      ),
+
+      // BCMI
+      bcmiDescription: new FormControl(
+        // default to using the master description if the flavour record does not exist
+        (this.currentRecord &&
+          ((this.bcmiFlavour && this.bcmiFlavour.description) ||
+          (!this.bcmiFlavour && this.currentRecord.description))) ||
+          ''
+      ),
+      publishBcmi: new FormControl({
+        value: (this.currentRecord && this.bcmiFlavour && this.bcmiFlavour.read.includes('public')) || false,
+        disabled: !this.factoryService.userInBcmiRole()
+      })
+    });
+  }
+
+  navigateToDetails() {
+    this.router.navigate(['records', 'annual-reports', this.currentRecord._id, 'detail']);
```

Comment:
Without the `/` in front of records this will navigate relative the route it is called from. Is that intended? Depending on where this screen is it might work as expected without the `/` too.

Reply:
For the routers here I just copied it exactly from the other details/add-edit screens. Happy to make a change, but if so I should make the change to all of them, not just the five new ones... they all seem to work as expected for now though!